CoreFunctions.js

```diff
@@ -844,3 +844,125 @@ function areArraysEqualSets(a1, a2) {
 
   return true;
 }
+
+
+function find_or_create_generic_draggable_window(id, titleBarText, addLoadingIndicator = true, addPopoutButton = false) {
+  console.log(`find_or_create_generic_draggable_window id: ${id}, titleBarText: ${titleBarText}, addLoadingIndicator: ${addLoadingIndicator}, addPopoutButton: ${addPopoutButton}`);
+  const existing = id.startsWith("#") ? $(id) : $(`#${id}`);
+  if (existing.length > 0) {
+    return existing;
+  }
+
+  const container = $(`<div class="resize_drag_window" id="${id}"></div>`);
+  container.css({
+    "left": "10%",
+    "top": "10%",
+    "max-width": "100%",
+    "max-height": "100%",
+    "position": "fixed",
+    "height": "80%",
+    "width": "80%",
+    "z-index": "10000",
+    "display": "none"
+  });
+
+  $("#site").append(container);
+
+  if (addLoadingIndicator) {
+    container.append(build_combat_tracker_loading_indicator(`Loading ${titleBarText}`));
+    const loadingIndicator = container.find(".sidebar-panel-loading-indicator");
+    loadingIndicator.css("top", "25px");
```

Comment:
nit: I would stack these for painting efficiency. `loadingIndicator.css({"top": "25px", "height": "calc(100% - 25px)"});